src/librbd/io/ImageRequestWQ.h

```diff
@@ -107,6 +108,25 @@ class ImageRequestWQ
   std::atomic<unsigned> m_io_blockers { 0 };
   std::atomic<unsigned> m_io_throttled { 0 };
 
+  typedef interval_set<uint64_t> ImageExtents;
+
+  ImageExtents m_in_flight_extents;
+
+  vector<ImageDispatchSpec<ImageCtxT>*> m_blocked_ios;
+  ceph::mutex m_ordering_lock;
+
+  bool block_overlapping_io(
+    ImageExtents* in_flight_image_extents, uint64_t object_off,
+    uint64_t object_len);
+
+  void unblock_overlapping_io(uint64_t off, uint64_t len, uint64_t tid);
```

Comment:
Nit: pass `ImageDispatchSpec` instead of individual params

Reply:
This doesn't seem possible since after `req->send()` in `process()`, the image extents are not preserved. I'll probably have to hold onto a copy of `ImageDispatchSpec` before `req->send()` in order to achieve this. I think that would be ok?

Reply:
Ack. I guess for this particular method it would make sense to pass the extents (just add a comment to explain that the extents are invalidated after the request is sent so that the purpose is clear).

Reply:
Done. Added comment.